modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/ScriptParseResult.java

```diff
@@ -69,25 +80,33 @@ public List<StatementParseResult> results() {
     }
 
     /**
-     * Counts the number of {@link SqlDynamicParam} nodes in the tree.
+     * Adjusts the dynamic parameter indexes to match the single statement parameter indexes.
      */
     @NotThreadSafe
-    static class SqlDynamicParamsCounter extends SqlBasicVisitor<Object> {
-        int count;
+    private static final class SqlDynamicParamsAdjuster extends SqlShuttle {
+        private int counter;
 
         @Override
-        public @Nullable Object visit(SqlDynamicParam param) {
-            count++;
-
-            return null;
+        public SqlNode visit(SqlDynamicParam param) {
+            return new SqlDynamicParam(counter++, param.getParserPosition());
         }
 
-        int forNode(SqlNode node) {
-            count = 0;
+        /**
+         * Rewrites dynamic parameter nodes in the provided tree.
```

Comment:
I guess, this javadoc describes prev. method `visit(SqlDynamicParam param)`.
This method resets state between processing different `StatementParseResult` and allow to reuse SqlDynamicParamsAdjuster, isn't it?

Can you drop this method and write a new one that will reset state explicitly?
`void reset() { counter = 0; }`
then fix SqlDynamicParamsAdjuster usage: 
```
   dynamicParamsAdjuster.reset();
   SqlNode newTree = dynamicParamsAdjuster.visitNode(node);
```

Reply:
Done.